nucliadb/nucliadb/search/requesters/utils.py

```diff
@@ -196,7 +206,9 @@ def validate_node_query_results(results: list[Any]) -> Optional[HTTPException]:
                     logger.warning(
                         f"GRPC error while querying shard data: {result.debug_error_string()}"
                     )
-                    status_code = 503
+                    for node in used_nodes:
```

Comment:
When querying multiple logical shards, this can reconnect healthy connections, as it doesn't check which one have failed

Reply:
Alright, pushed update, lmkwyt